### DISCUSSION TITLE
Rendering mesh as wireframe

### DISCUSSION THREAD

**Discussion by bast** [emilk/egui#1450] [Q&A] [answered]
First of all thanks for this wonderful software!

I am interested in creating a wireframe mesh using https://docs.rs/egui/latest/egui/struct.Mesh.html
I am able to render a mesh where triangles are filled with a color or colors. What I would like to do instead is that triangles are transparent (or have a solid fill color) and where the wireframe is visible? In other words I would like to render the triangle edges/sides in a mesh instead or in addition to only fill the triangles with color but not seeing their "border".

Any pointer of where to look in documentation or demos or source code would be super appreciated.

Thank you!

**Comment by emilk** [comment, marked as answer]
You can create a `Shape::LineSegment` for each edge in your triangle mesh!

**Reply by bast** [reply]
Thank you! I will test this out.

**Reply by bast** [reply]
I have tried it with `Shape::LineSegment` and this indeed works. However, I have over a million triangles and whereas the `Mesh` is very fast and snappy when zooming and drag and dropping at this problem size, adding the wire frame via `LineSegment` slows the UI down a lot. I have 1.5 M line segments. Is this slow down expected when comparing Mesh and LineSegment? Can I solve this with a custom shader for the Mesh? (just trying to find out whether I am on the right track, I can then piece details together)

**Reply by emilk** [reply]
You should probably look at some 3D library to accomplish what you want, e.g. `bevy` or `three-d`. `egui/epaint` is designed for simpler things.

**Reply by bast** [reply]
Thank you for the clarification.

**Reply by bast** [reply]
I solved it by switching on line segments only after certain zoom level, combined with clipping segments that are not in view. It looks good and it's back to super snappy. Thanks again for the pointers!

**Reply by bast** [reply]
Amazingly cool library. I am glad I can continue in egui and don't have to switch.
